### PR TITLE
fixed #7564: Recording creates a track but UI does not update

### LINKED ISSUE #7564: Recording creates a track but UI does not update
Steps to reproduce:
1. Create an empty project
2. Press 'record'
3. Observe

What's happening: the track is created under the hood, recording is saved but UI is not updating at this point at all. To be able to see that it's true, save project, close it and open it again -> recorded clip on a new track will appear.

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #7564

### CODE REVIEW

File: src/record/internal/au3/au3record.cpp [hunk 1]
```diff
@@ -433,193 +410,167 @@ Ret Au3Record::doRecord(Au3Project& project,
...
+        // recording to NEW track(s).
+        bool recordingNameCustom, useTrackNumber, useDateStamp, useTimeStamp;
+        wxString defaultTrackName, defaultRecordingTrackName;
+
+        // Count the tracks.
+        auto numTracks = trackList.Any<const Au3WaveTrack>().size();
+
+        auto recordingChannels = std::max(1, AudioIORecordChannels.Read());
+
+        gPrefs->Read(wxT("/GUI/TrackNames/RecordingNameCustom"), &recordingNameCustom, false);
```
Comment (igorkorsukov) [inline src/record/internal/au3/au3record.cpp:488]: TODO, replace with configuration

File: src/record/internal/au3/au3record.cpp [hunk 2]
```diff
@@ -433,193 +410,167 @@ Ret Au3Record::doRecord(Au3Project& project,
...
+                }
+                if (t1 < DBL_MAX) {
+                    t1 = newTrack->SnapToSample(t1);
                 }
+            }
 
-                newTrack->MoveTo(t0);
-                wxString nameSuffix = wxString(wxT(""));
+            newTrack->MoveTo(t0);
+            wxString nameSuffix = wxString(wxT(""));
```
Comment (igorkorsukov) [inline src/record/internal/au3/au3record.cpp:513]: TODO, move the name formation to a separate function

File: src/record/internal/au3/au3record.cpp [hunk 3]
```diff
@@ -433,193 +410,167 @@ Ret Au3Record::doRecord(Au3Project& project,
...
+            m_recordData.clipsKeys.push_back({ newTrack->GetId(), newClip->GetId() });
+
+            trackedit::ITrackeditProjectPtr prj = globalContext()->currentTrackeditProject();
+            prj->notifyAboutTrackAdded(DomConverter::track(newTrack));
         }
+    }
 
-        int token
-            =gAudioIO->StartStream(transportSequences, t0, t1, t1, options);
+    int token = gAudioIO->StartStream(transportSequences, t0, t1, t1, options);
```
Comment (igorkorsukov) [inline src/record/internal/au3/au3record.cpp:561]: TODO, use IAudioEngine (add methods to it if necessary)

**Review by kryksyh — APPROVED** [review]
Sorry for delay :)

**Pull request merged**